### PR TITLE
Fix download condition for JavaFX static libs

### LINKED ISSUE #489: [Linux] JavaFX static libs are downloaded everytime compile is executed
[Present condition](https://github.com/gluonhq/substrate/blob/c9b5dcaa54d8733fdcbd2c7af58baf34b963b7db/src/main/java/com/gluonhq/substrate/util/FileDeps.java#L236) fails for android linux static libs as they come packaged with `libglass_monocle.a` instead of [`libglass.a`](https://github.com/gluonhq/substrate/blob/c9b5dcaa54d8733fdcbd2c7af58baf34b963b7db/src/main/java/com/gluonhq/substrate/util/FileDeps.java#L59)

### PR DESCRIPTION
**Pull request by abhinayagarwal**
Fixes #489

### CODE REVIEW

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/gluonhq/substrate/util/FileDeps.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=5297576&bid=17340315)

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/gluonhq/substrate/util/FileDeps.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=5297576&bid=17340315)

File: src/main/java/com/gluonhq/substrate/util/FileDeps.java [hunk 1]
```diff
@@ -231,10 +234,9 @@ private boolean setupDependencies() throws IOException {
                 downloadJavaFXStatic = true;
             } else {
                 String path = javafxStatic.toString();
-                if (JAVAFX_FILES.stream()
-                        .map(s -> new File(path, s))
-                        .anyMatch(f -> !f.exists())) {
-                    Logger.logDebug("jar file not found");
+                if (JAVAFX_FILES.stream().map(s -> new File(path, s)).anyMatch(f -> !f.exists()) &&
```
Comment (jperedadnr) [inline src/main/java/com/gluonhq/substrate/util/FileDeps.java:237]: if the check for the JavaFX jars is true, means that some jar is missing, therefore: `downloadJavaFXStatic = true;`
Adding "and either libglassXX.a"  doesn't help. Shouldn't it be `noneMatch`?

```
if (JAVAFX_FILES.stream().map(s -> new File(path, s)).anyMatch(f -> !f.exists()) || 
     JAVAFX_STATIC_FILES.stream().map(s -> new File(path, s)).noneMatch(File::exists)) {
```

Comment (abhinayagarwal) [reply src/main/java/com/gluonhq/substrate/util/FileDeps.java:237]: Have you checked #489 for explanation?

Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/util/FileDeps.java:237]: Yes, I did. 
We used the javafx list to verify that those files are available or not. If any of them is missing, we download.
Now we need to split that in two: jars and libs. Okay.

But your code is doing:
 - if any jar is missing and either lib exists -> download

so at this point, if a jar is missing but one lib exist, it won't download. Or if all jars exist, but no lib is found, it won't download.

Imo, the check should be:
- if any jar is missing -> download, or
- if none of these libs is found -> download

Comment (abhinayagarwal) [reply src/main/java/com/gluonhq/substrate/util/FileDeps.java]: Makes sense. Thank you for the explanation.

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/gluonhq/substrate/util/FileDeps.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=5297576&bid=17340315)

**Pull request merged**